omegaconf/omegaconf.py

```diff
@@ -906,6 +906,9 @@ def _node_wrap(
     )
     if is_dict:
         key_type, element_type = get_dict_key_value_types(type_)
+        flags = {}
+        if is_dict_annotation(type_):
```

Comment:
I'm a bit confused with the distinction between `type_` and `ref_type` in this function: they are both set to the same value in the code calling it, but tests seem to use different values sometimes. Also in this function there are both occurrences of `ref_type=type_` and `ref_type=ref_type`, which doesn't help understanding what is going on.

Would it be possible to clarify in a docstring or something?

Reply:
Both `type_` and `ref_type` are constant within this function's body.
~We can refactor by replacing each `type_` with `ref_type` and dropping `type_` from the function signature.~
Nevermind, @odelalleau you make a good point about the tests calling with different values.

Reply:
This is complicated and the desired behavior is driven by the tests.
At a high level ref_type is equivalent to an annotation type and object_type is, well, the object type.

```python
x : Any = "str"
```
In the above, the ref_type of x would be Any and the object_type would be str.
ref_type is usually used when assigning an object:
```python
x : str = "str"
x = 10 # will trigger a validation error.

# in actual OmegaConf code:
cfg = OmegaConf.create({"x" : StringNode("str")})
cfg.x = 10

# or with dataclass:
@dataclass
class Foo:
  x : str = "str"

cfg = OmegaConf.structured(Foo)
cfg.x = 10
```

There are some scenarios, especially during merge - where we use the ref_type for more things.